src/Payload.cpp

```diff
@@ -942,7 +996,10 @@ omnmmsgPayload_createFromByteBuffer (msgPayload*         msg,
 {
     if (NULL == msg || NULL == buffer) return MAMA_STATUS_NULL_ARG;
     if (0 == bufferLength) return MAMA_STATUS_INVALID_ARG;
-    omnmmsgPayload_create (msg);
+
+    OmnmPayloadImpl* impl = new OmnmPayloadImpl(bufferLength);
```

Comment:
why?

Reply:
I assume as opposed to just delegating to `omnmmsgPayload_create` (which does the same thing)?

Reply:
yeah - not sure the value of the change

Reply:
Ahh... in both cases we know the size of the buffer so are able to allocate proper size right off.